src/repositories/request-repository.ts

```diff
@@ -0,0 +1,80 @@
+import CID from "cids";
+import { EntityRepository, InsertResult, UpdateResult } from "typeorm";
+import { BaseRepository } from 'typeorm-transactional-cls-hooked';
+
+import Context from "../context";
+import Contextual from "../contextual";
+import { Request, RequestUpdateFields } from "../models/request";
+import { RequestStatus } from "../models/request-status";
+import { config } from "node-config-ts";
+
+@EntityRepository(Request)
+export default class RequestRepository extends BaseRepository<Request> implements Contextual {
```

Comment:
ditto re: adding a comment.  It's not immediately clear to me what data lives in the RequestRepository and what data lives in the AnchorRepository, so it'd be nice if the comments for these classes could clarify that as well

Reply:
It's pretty straightforward to conclude what data "lives" in repositories by looking at the `@EntityRepository(Entity)` annotation. Here's the [link](https://typeorm.io/#/custom-repository) for more information.

Reply:
Well, I already spoke above about my feelings on requiring an understanding of all imported libraries in a file in order to understand the code in the file itself, but that actually wasn't what I was even trying to get at with this comment as I had already brought that up above.  Apologies for not being clearer about the difference between I was getting at with this comment vs what I was saying above!  

With this comment I was trying to actually point to ambiguity not around what TypeORM is or is doing, but around what the `Request` and `Anchor` objects represent.  They are ORM Model objects that can be serialized/reconstructed from the database, but what's missing imo is a description of what concept in our code they actually represent.  My understanding is that when the request-controller's 'create' endpoint is activated, a `Request` object is created and persisted into the database.  It then seems that when we periodically perform an anchor, we query for pending Requests from the database and build up `Anchor` objects that in turn get persisted to the database.  It would be good for these basic descriptions of how these objects are used and what they represent to be written down somewhere, presumably in some combination of comments on the `Request` and `Anchor` classes, and comments on the `RequestRepository` and `AnchorRepository` classes.  
Additionally, some other questions that would be nice to answer in such comments would include:
* What's the relationship between a `Request` and an `Anchor`?  Are they one-to-one, or does one `Anchor` record contain information about all the `Requests` that got bundled into a single merkle tree anchor blockchain transaction?
* What's the relationship between an `Anchor` record and the anchor transactions that get created against the blockchain?  Are the `Anchor` records persisted to the database before or after the corresponding transactions are submitted on chain?
* What is the lifetime of both of these records?  Are they garbage collected ever, or do they live in the database forever in a 'complete' state?

Overall I just feel like more comments explaining the workflow, lifecycle, and relationship between the various objects, concepts, and services that are used, implemented, or interacted with in this repository would make it much easier for someone new to get up to speed on this code base.

Reply:
I get what you're saying. Can we open another story (or stories) for CAS comments and documentation since this PR is blocked and it actually solves another set of problems which are really urgent and need to happen in this sprint?